Lib/test/test_descr.py

```diff
@@ -3900,6 +3900,18 @@ def __ipow__(self, other):
         a = C()
         a **= 2
 
+    def test_ipow_returns_not_implemented(self):
+        class A:
+            def __ipow__(self, other):
```

Comment:
It looks like we're also lacking a test for when `A` doesn't define `__ipow__`. Do you mind adding one?

Reply:
Sure, added the test

Reply:
Could you add a test for the two fallbacks and ensure the same class is returning `NotImplemented` and implementing the regular power methods i.e something like:

```python
class A:
    def __ipow__(self, other):
        return NotImplemented

class B(A):
    def __pow__(self, other):
        return 1
class C(A):
    def __rpow__(self, other):
        return 2
```

The linked bug is only exhibited when a class has an ipow slot but when called it returns `NotImplemented`.

Reply:
Not sure I understood correctly what you meant, but I updated the test to also test the `__pow__` fallback.

Reply:
Thank you, what you have now looks good. What I meant was that to catch the original bug/properly be a regression test, the class has to look like:

```python
class A:
    def __ipow__(self, _):
        return NotImplemented
    def __pow__(self, other):
        ...
```

if the class looks like

```python
class A:
    def __pow__(self, other):
        ...
```
then the old logic returns a false here and correctly goes to the non-inplace version of power anyway: https://github.com/python/cpython/blob/8e19c8be87017f6bef8e4c936b1e6ddacb558ad2/Objects/abstract.c#L1162-L1163

Reply:
Ah yes, you are right. Thanks!